bson/element.go

```diff
@@ -400,5 +400,18 @@ func convertValueToElem(key string, v *Value) *Element {
 	elem := newElement(0, uint32(keyLen+2))
 	elem.value.data = d
 
+	if v.d == nil {
+		return elem
+	}
+	mdata, err := v.d.MarshalBSON()
```

Comment:
I see that we're marshaling and then immediately unmarshaling the document here. Is there a particular reason you wrote it this way?

Reply:
Well, I thought that it would be better to make a deep copy here, but was unable to find something to do that on document, so I did that through marshaling.
If it's ok to go with pointer to original doc - I'm ok with that :)